### PR TITLE
Fix fourmolu

### PR DESCRIPTION
**Pull request by brandonchinn178**
---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.
* [ ] Include [manual QA notes](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#qa-notes) if your PR relates to cabal-install.

Bonus points for added automated tests!

### CODE REVIEW

**Review by ulysses4ever — APPROVED** [review]
Thanks!

I'd like @Kleidukos to take a look, if possible.

**Comment by ulysses4ever** [comment]
Also, this is funny: I can swear we fixed the capitalization once...

**Comment by ulysses4ever** [comment]
Ah, it was never merged. Should be closed now:  #8994

**Comment by brandonchinn178** [comment]
Sorry, Lexer.hs linting is broken. I'll fix it in a couple hours (I've stepped away for a bit)

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -15,9 +15,4 @@ jobs:
         pattern: |
           Cabal/**/*.hs
           Cabal-syntax/**/*.hs
-          Cabal-install/**/*.hs
-          !Cabal-syntax/src/Distribution/Fields/Lexer.hs
-          !Cabal-syntax/src/Distribution/SPDX/LicenseExceptionId.hs
-          !Cabal-syntax/src/Distribution/SPDX/LicenseId.hs
-          !Cabal/src/Distribution/Simple/Build/Macros/Z.hs
-          !Cabal/src/Distribution/Simple/Build/PathsModule/Z.hs
```
Comment (Kleidukos) [inline .github/workflows/linting.yml:23]: Thanks a bunch for removing this dreadful hack :)

File: fourmolu.yaml [hunk 2]
```diff
@@ -7,8 +7,13 @@ respectful: true # don't be too opinionated about newlines etc.
...
 function-arrows: leading
 single-constraint-parens: never
 in-style: right-align
 let-style: auto
+
+fixities:
+  # Distribution.FieldGrammar
+  - infixl 5 ^^^
+  # Distribution.Types.InstalledPackageInfo.FieldGrammar
+  - infixl 4 <@>
```
Comment (Kleidukos) [inline fourmolu.yaml:19]: Fantastic!

**Review by Kleidukos — APPROVED** [review]

**Comment by ulysses4ever** [comment]
Currently, GitHub says:

> This branch has conflicts that must be resolved

**Review by geekosaur — APPROVED** [review]

File: fourmolu.yaml [hunk 3]
```diff
@@ -19,3 +19,9 @@ fixities:
   - infixl 5 ^^^
   # Distribution.Types.InstalledPackageInfo.FieldGrammar
   - infixl 4 <@>
+
+reexports:
```
Comment (brandonchinn178) [inline fourmolu.yaml:23]: @ffaf1 I went ahead and added the config here, to facilitate https://github.com/fourmolu/fourmolu/pull/346. The styling is still fourmolu 0.12, but fourmolu 0.12 will ignore this config

**Comment by andreabedini** [comment]
Thank you @brandonchinn178 !! <3

**Comment by brandonchinn178** [comment]
@Kleidukos who should hit the big green button?

**Comment by ulysses4ever** [comment]
@brandonchinn178 the bot https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#github-pull-request-conventions

**Pull request merged**